src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.ShoppingCartDao;
+import com.cinema.exeption.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.User;
+import com.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add shopping cart with id "
+                    + shoppingCart.getId(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> getCartByUserQuery
+                    = session.createQuery("FROM ShoppingCart sc left join fetch sc.tickets "
+                    + "WHERE sc.user = : user", ShoppingCart.class);
+            getCartByUserQuery.setParameter("user", user);
```

Comment:
do you have access to user field of this shopping cart when the session is closed?

Reply:
Done

Reply:
Now I have